docs/_docs/02_testing-best-practices/testing-plan-output.md

```diff
@@ -0,0 +1,40 @@
+---
+layout: collection-browser-doc
+title: Testing plan output
+category: testing-best-practices
+excerpt: >-
+  Testing specific changes described in Terraform plan output
+tags: ["testing-best-practices"]
+order: 202
+nav_title: Documentation
+nav_title_link: /docs/
+---
+
+Sometimes you can get the information you need in your tests from a `terraform plan`; you may not need to do an `apply`.  Terratest provides a way to use certain information from a `terraform plan` in your tests.
+
+The `InitAndPlanWithInfo` function will return a `PlanInfo` struct that will contain information about which resources would be changed and what specifically about them would change.
```

Comment:
Is there a way to make this work with the output from `InitAndPlan`? Or is the new function absolutely necessary because of how you need to extract the information?

Reply:
I think we do need a new function.  The only way that I can see to get the JSON representation of the plan is to save the plan output to a file with `terraform plan -o some.plan` and then run `terraform show -json some.plan` on that file.

Reply:
Maybe this new function could use the current plan functions and pass an `-out` arg to the `RunTerraformComand` functions with the path to a tmp file, and then maybe there would be separate `Show` functions, and `InitAndPlanWithInfo` would call the relevant `Plan` then `Show` functions and unmarshall the result from the `Show` function.

Reply:
Ah I see. Given that machinery I think having the `InitAndPlanWithInfo` as a helper would be good, but eventually we can probably break out each step:

```
InitAndPlan  // Update Options so that you can have it store a plan file
planInfo := ShowPlanInfo(planfile) // takes the plan file and returns the info
```

The nice thing about the second function is that you can actually inject that into a CI/CD pipeline for production code to smoke check changes!